customfieldnames.civix.php

```diff
@@ -346,3 +346,166 @@ function _customfieldnames_civix_civicrm_alterSettingsFolders(&$metaDataFolders
     $metaDataFolders[] = $settingsDir;
   }
 }
+
+/**
+ * Implements hook_civicrm_buildForm().
+ *
+ * @param string $formName
+ * @param CRM_Core_Form $form
+ */
+function _customfieldnames_civix_civicrm_buildForm($formName, &$form) {
+  $templatePath = realpath(dirname(__FILE__)."/templates");
+  if( ($formName == 'CRM_Custom_Form_Group') && ($form->getAction() == CRM_Core_Action::ADD) ) {
+    $templatePath = realpath(dirname(__FILE__)."/templates");
+    $form->add('text', 'table_name', ts('Set Table Name'), '', FALSE);
+    $form->assign('table_name', TRUE);
+    $form->add('text', 'name', ts('Set Group Name'), '', FALSE);
+    $form->assign('name', TRUE);
+    
+    CRM_Core_Region::instance('page-body')->add(array(
+      'template' => "{$templatePath}/customgroup.tpl"
+    ));
+  }
+  if ( ($formName == 'CRM_Custom_Form_Field') && ($form->getAction() == CRM_Core_Action::ADD) ) {
+    $form->add('text', 'column_name', ts('Set Column Name'), '', FALSE);
+    $form->assign('column_name', TRUE);
+    
+    $form->add('text', 'name', ts('Set Field Name'), '', FALSE);
+    $form->assign('name', TRUE);
+    
+    CRM_Core_Region::instance('page-body')->add(array(
+      'template' => "{$templatePath}/customgroupfield.tpl"
+    ));
+  }
+}
+
+/**
+ * Implements hook_civicrm_validateForm().
+ *
+ * @param string $formName
+ * @param array $fields
+ * @param array $files
+ * @param CRM_Core_Form $form
+ * @param array $errors
+ */
+function _customfieldnames_civix_civicrm_validateForm($formName, &$fields, &$files, &$form, &$errors) {
```

Comment:
change label references to match changes above throughout this function

Reply:
Done, also as the empty field validation has removed so no need to set field label in this function.